lib/_stream_readable.js

```diff
@@ -434,7 +432,7 @@ Readable.prototype.read = function(n) {
   debug('need readable', doRead);
 
   // if we currently have less than the highWaterMark, then also read some
-  if (state.length === 0 || state.length - n < state.highWaterMark) {
+  if (state.length - n < state.highWaterMark) {
```

Comment:
Please note that `state.highWaterMark`, `state.length` and `n` maybe equal 0. So if they all equal 0, `state.length - n < state.highWaterMark` doesn't equal to `state.length === 0 || state.length - n < state.highWaterMark`.

I suggest that this change would be with another PR because **code cleaning shouldn't change code logic**.

Reply:
Nice catch @MoonBall